Mathlib/Topology/UniformSpace/CompactConvergence.lean

```diff
@@ -391,4 +390,29 @@ instance instCompleteSpaceOfSequentialSpace [SequentialSpace α] :
 
 end CompleteSpace
 
+/-- If `C(α, β)` is a complete space, then for any (possibly, discontinuous) function `f`
+and any set `s`, the set of functions `g : C(α, β)` that are equal to `f` on `s`
+is a complete set.
+
+Note that this set does not have to be a closed set.
+This lemma is useful to prove that, e.g., the space of paths between two points
+and the space of homotopies between two continuous maps are complete spaces,
+without assuming that the codomain is a Hausdorff space. -/
+theorem isComplete_setOf_eqOn [CompleteSpace C(α, β)] (f : α → β) (s : Set α) :
+    IsComplete {g : C(α, β) | EqOn g f s} := by
+  classical
+  intro l hlc hlf
+  rcases CompleteSpace.complete hlc with ⟨f', hf'⟩
+  have := hlc.1
+  have H₁ : ∀ x ∈ s, Inseparable (f x) (f' x) := fun x hx ↦ by
+    refine tendsto_nhds_unique_inseparable ?_ ((continuous_eval_const x).continuousAt.mono_left hf')
+    refine tendsto_const_nhds.congr' <| .filter_mono ?_ hlf
+    exact fun _ h ↦ (h hx).symm
+  have H₂ (x) : Inseparable (s.piecewise f f' x) (f' x) := by
+    by_cases hx : x ∈ s <;> simp [hx, H₁, Inseparable.refl]
+  set g : C(α, β) :=
+    ⟨s.piecewise f f', (continuous_congr_of_inseparable H₂).mpr <| map_continuous f'⟩
+  refine ⟨g, Set.piecewise_eqOn _ _ _, hf'.trans_eq ?_⟩
+  rwa [eq_comm, ← Inseparable, ← inseparable_coe, inseparable_pi]
```

Comment:
Would the following be easier?

- use [ContinuousMap.isUniformInducing_comp](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/UniformSpace/CompactConvergence.html#ContinuousMap.isUniformInducing_comp) to get that the obvious map `C(α, β) → C(α, SeparationQuotient β)` is a uniform inducing
- show that the image of your set is precisely `{g : C(α, SeparationQuotient β) | EqOn g (SeparationQuotient.mk \circ f) s}` using the same piecewise arguments
- conclude by [IsUniformInducing.isComplete_iff](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/UniformSpace/UniformEmbedding.html#IsUniformInducing.isComplete_iff)

Reply:
I've started this, but it doesn't seem to be much shorter/simpler. I would prefer to merge this proof, then update it in another PR.